src/amo/components/AddonReviewListItem/index.js

```diff
@@ -54,6 +58,20 @@ type InternalProps = {|
 |};
 
 export class AddonReviewListItemBase extends React.Component<InternalProps> {
+  onClickToDeleteReview = (event: SyntheticEvent<HTMLElement>) => {
+    const { dispatch, errorHandler, isReplyToReviewId, review } = this.props;
+    event.preventDefault();
+
+    invariant(review, 'review is required');
+    dispatch(
+      deleteAddonReview({
```

Comment:
I just noticed that when this action dispatches, the UI does not indicate that anything is happening. This could be confusing to users on a slow connection. In fact, they may think nothing is happening and they may try to delete the review again.

Here's a simple idea for a fix:

<img width="573" alt="screenshot 2018-08-29 10 31 05" src="https://user-images.githubusercontent.com/55398/44798403-0483bb00-ab77-11e8-98e6-80ba4d01a002.png">


<details>
<summary>patch</summary>

```diff
diff --git a/src/amo/components/AddonReviewListItem/index.js b/src/amo/components/AddonReviewListItem/index.js
index eb78c9925..2dd647224 100644
--- a/src/amo/components/AddonReviewListItem/index.js
+++ b/src/amo/components/AddonReviewListItem/index.js
@@ -47,6 +47,7 @@ type Props = {|
 
 type InternalProps = {|
   ...Props,
+  deletingReview: boolean,
   dispatch: DispatchFunc,
   editingReview: boolean,
   errorHandler: ErrorHandlerType,
@@ -199,6 +200,7 @@ export class AddonReviewListItemBase extends React.Component<InternalProps> {
   render() {
     const {
       addon,
+      deletingReview,
       editingReview,
       errorHandler,
       i18n,
@@ -254,26 +256,32 @@ export class AddonReviewListItemBase extends React.Component<InternalProps> {
                 ? i18n.gettext('Edit my reply')
                 : i18n.gettext('Edit my review')}
             </a>
-            <ConfirmButton
-              buttonType="cancel"
-              cancelButtonType="neutral"
-              className={makeClassName(
-                'AddonReviewListItem-control',
-                confirmButtonClassName,
-              )}
-              confirmButtonText={i18n.gettext('Delete')}
-              id={`${confirmButtonClassName}-${review.id}`}
-              message={
-                isReply
-                  ? i18n.gettext('Do you really want to delete this reply?')
-                  : i18n.gettext('Do you really want to delete this review?')
-              }
-              onConfirm={this.onClickToDeleteReview}
-            >
-              {isReply
-                ? i18n.gettext('Delete my reply')
-                : i18n.gettext('Delete my review')}
-            </ConfirmButton>
+            {deletingReview ? (
+              <span className="AddonReviewListItem-control">
+                {i18n.gettext('Deleting…')}
+              </span>
+            ) : (
+              <ConfirmButton
+                buttonType="cancel"
+                cancelButtonType="neutral"
+                className={makeClassName(
+                  'AddonReviewListItem-control',
+                  confirmButtonClassName,
+                )}
+                confirmButtonText={i18n.gettext('Delete')}
+                id={`${confirmButtonClassName}-${review.id}`}
+                message={
+                  isReply
+                    ? i18n.gettext('Do you really want to delete this reply?')
+                    : i18n.gettext('Do you really want to delete this review?')
+                }
+                onConfirm={this.onClickToDeleteReview}
+              >
+                {isReply
+                  ? i18n.gettext('Delete my reply')
+                  : i18n.gettext('Delete my review')}
+              </ConfirmButton>
+            )}
           </React.Fragment>
         ) : null}
 
@@ -323,18 +331,21 @@ export class AddonReviewListItemBase extends React.Component<InternalProps> {
 }
 
 export function mapStateToProps(state: AppState, ownProps: Props) {
+  let deletingReview = false;
   let editingReview = false;
   let replyingToReview = false;
   let submittingReply = false;
   if (ownProps.review) {
     const view = state.reviews.view[ownProps.review.id];
     if (view) {
+      deletingReview = view.deletingReview;
       editingReview = view.editingReview;
       replyingToReview = view.replyingToReview;
       submittingReply = view.submittingReply;
     }
   }
   return {
+    deletingReview,
     editingReview,
     siteUserHasReplyPerm: hasPermission(state, ADDONS_EDIT),
     replyingToReview,
diff --git a/src/amo/reducers/reviews.js b/src/amo/reducers/reviews.js
index 240b26304..87070a35f 100644
--- a/src/amo/reducers/reviews.js
+++ b/src/amo/reducers/reviews.js
@@ -2,6 +2,7 @@
 import { oneLine } from 'common-tags';
 
 import {
+  DELETE_ADDON_REVIEW,
   FLASH_REVIEW_MESSAGE,
   HIDE_FLASHED_REVIEW_MESSAGE,
   UNLOAD_ADDON_REVIEWS,
@@ -22,6 +23,7 @@ import {
   createInternalReview,
 } from 'amo/actions/reviews';
 import type {
+  DeleteAddonReviewAction,
   UnloadAddonReviewsAction,
   FlagReviewAction,
   FlashMessageType,
@@ -75,6 +77,7 @@ export type FlagState = {
 };
 
 type ViewStateByReviewId = {|
+  deletingReview: boolean,
   editingReview: boolean,
   flag: FlagState,
   replyingToReview: boolean,
@@ -175,6 +178,7 @@ export const changeViewState = ({
     view: {
       ...state.view,
       [reviewId]: {
+        deletingReview: false,
         editingReview: false,
         replyingToReview: false,
         submittingReply: false,
@@ -270,6 +274,7 @@ export const addReviewToState = ({
 };
 
 type ReviewActionType =
+  | DeleteAddonReviewAction
   | FlagReviewAction
   | FlashReviewMessageAction
   | UnloadAddonReviewsAction
@@ -296,6 +301,12 @@ export default function reviewsReducer(
   }: {| _addReviewToState: typeof addReviewToState |} = {},
 ) {
   switch (action.type) {
+    case DELETE_ADDON_REVIEW:
+      return changeViewState({
+        state,
+        reviewId: action.payload.reviewId,
+        stateChange: { deletingReview: true },
+      });
     case SEND_REPLY_TO_REVIEW:
       return changeViewState({
         state,
```

</details>

What do you think?

Reply:
I like it! It's pretty convenient that we don't need a `finishedDeleting` action because the delete process itself will clear the state of the `deletingReview` flag. 👍